R/import_mzroll.R

```diff
@@ -168,6 +168,110 @@ process_mzroll <- function(mzroll_db_path,
+#' 
+#' @export
+process_metabolon <- function(peak_areas_path,
+                              chem_anno_path,
+                              metadata_path,
+                              peak_floor_intensity = 4096
+                              ) {
+  
+  checkmate::assertFileExists(peak_areas_path)
+  checkmate::assertFileExists(chem_anno_path)
+  checkmate::assertFileExists(metadata_path)
+  checkmate::assertNumeric(peak_floor_intensity)
+  
+  # import metabolon results as CSV files
+  peaks <- read.csv(file=peak_areas_path, check.names = FALSE)
+  anno <- read.csv(file=chem_anno_path)
+  metadata <- read.csv(file=metadata_path)
+  
+  columns_to_pivot <- colnames(peaks)
+  columns_to_pivot <- columns_to_pivot[2:length(columns_to_pivot)]
+  samples <- metadata %>%
+    select(PARENT_SAMPLE_NAME,CLIENT_SAMPLE_NUMBER) %>%
+    rename(sampleName=PARENT_SAMPLE_NAME,sampleId=CLIENT_SAMPLE_NUMBER) %>%
+    mutate(sampleId = factor(sampleId, levels = sampleId))
+  
+  peaks_updated_noIds <- peaks %>% 
+    pivot_longer(cols = all_of(columns_to_pivot)) %>% 
+    mutate(groupId = as.integer(name),
+           sampleName = PARENT_SAMPLE_NAME,
+           peakAreaTop = value) %>% # claman is hard-coded to use peakAreaTop as quant method from mzrolldb files
+    select(-name, -PARENT_SAMPLE_NAME, -value) %>%
+    left_join(samples, by="sampleName") %>%
+    mutate(peakAreaTop = ifelse(is.na(peakAreaTop), peak_floor_intensity, peakAreaTop)) %>%
+    dplyr::group_by(groupId) %>%
+    dplyr::mutate(
+      log2_abundance = log2(peakAreaTop),
+      centered_log2_abundance = log2_abundance - mean(log2_abundance)) %>%
+    ungroup()
+  
+  peakId <- seq(length = nrow(peaks_updated_noIds))
+  peaks_updated <- cbind(peakId, peaks_updated_noIds)
+  
+  peakgroups <- anno %>%
+    select(CHEM_ID, CHEMICAL_NAME, SMILES, PLATFORM,TYPE) %>%
+    rename(
+      groupId=CHEM_ID,
+      compoundName=CHEMICAL_NAME,
+      smiles=SMILES,
+      method_tag=PLATFORM,
+    )
+  
+  peakgroups_updated <- peakgroups %>%
+    mutate(tagString = " ",
+           mz= 100, # dummy value
+           rt = 10, # dummy value
+           compoundDB = "Metabolon",
+           searchTableName = " ",
+           label = "g", # assume all Metabolon IDs are good
+           groupId=factor(groupId, levels=groupId)
+    )
+  
+  samples_updated<- samples %>%
+    rename(name=sampleName)
+  
+  peaks_updated$groupId <- factor(
+    peaks_updated$groupId,
+    levels = levels(peakgroups_updated$groupId)
+  )
+  
+  peaks_updated$sampleId <- factor(
+    peaks_updated$sampleId,
+    levels = levels(samples_updated$sampleId)
+  )
+  
+  # combine metabolon data into mzroll_list with added metadata
+  mzroll_list <- romic::create_triple_omic(
+    measurement_df = peaks_updated,
+    feature_df = peakgroups_updated,
+    sample_df = samples_updated,
+    feature_pk = "groupId",
+    sample_pk = "sampleId",
+    omic_type_tag = "mzroll"
+  )
+  
+  mzroll_list<-claman::merge_samples_tbl(
```

Comment:
don't use the claman namespace in the claman package

Reply:
Hey @shackett , thanks for the comments. I've made the changes you've requested.